kernel_test.go

```diff
@@ -0,0 +1,87 @@
+package cbpfc
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net"
+	"os"
+	"testing"
+	"time"
+	"unsafe"
+
+	"golang.org/x/net/bpf"
+	"golang.org/x/sys/unix"
+)
+
+// kernelBackend is a backend that runs cBPF in the kernel
+func kernelBackend(tb testing.TB, insns []bpf.Instruction, in []byte) XDPAction {
+	filter, err := bpf.Assemble(insns)
+	if err != nil {
+		tb.Fatal(err)
+	}
+
+	// Use a unix socket to test the filter
+	// This doesn't risk interfering with any other network traffic, doesn't require / add special
+	// headers (as would be the case if we used UDP for example) that the XDP tests don't deal with,
+	// and doesn't require any special permissions
+
+	// Temporary file for unix socket
+	temp, err := ioutil.TempFile("", "cbpfc_test")
+	if err != nil {
+		tb.Fatal(err)
+	}
+	path := temp.Name()
+	if err := os.Remove(path); err != nil {
+		tb.Fatal(err)
+	}
+
+	read, err := net.ListenUnixgram("unixgram", &net.UnixAddr{Name: path, Net: "unixgram"})
```

Comment:
You don't need a path, you can just make Linux assign a name in the abstract namespace:
```suggestion
	read, err := net.ListenUnixgram("unixgram", &net.UnixAddr{Name: "", Net: "unixgram"})
```

Reply:
missing Close

Reply:
Good idea, both fixed :)